### PR TITLE
cluster: save to the region cache when pending-peers or down-peers change

### LINKED ISSUE #3456: Abnormalities in TiKV may cause slow PD heartbeat processing
### What did you do?

* Some TiKVs are damaged

### What did you expect to see?

* The cluster is working normally

### What did you see instead?

* Region heartbeat processing is slow
* The region information is not updated in time
* A large number of scheduling operations failed

### What version of PD are you using (`pd-server -V`)?

* v4.0.10

### PR DESCRIPTION
**Pull request by HunDunDM**
### What problem does this PR solve?

close #3456

### What is changed and how it works?

* Before: save to the region cache when pending-peers or down-peers exists
* After: save to the region cache when pending-peers or down-peers change

### Check List

Tests

- Unit test

Related changes

- Need to cherry-pick to the release branch

### Release note

- save to the region cache when pending-peers or down-peers change

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -591,11 +591,13 @@ func (c *RaftCluster) processRegionHeartbeat(region *core.RegionInfo) error {
 			}
 			saveCache, needSync = true, true
 		}
-		if len(region.GetDownPeers()) > 0 || len(region.GetPendingPeers()) > 0 {
-			saveCache = true
+		if !core.SortedPeersStatsEqual(region.GetDownPeers(), origin.GetDownPeers()) {
+			log.Debug("down-peers changed", zap.Uint64("region-id", region.GetID()))
```
Comment (rleungx) [inline server/cluster/cluster.go:595]: How about printing the difference?

Comment (HunDunDM) [reply server/cluster/cluster.go:595]: Personally, I think pending is more common, so I used the `Debug` level.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**